abstutil/src/cli.rs

```diff
@@ -136,6 +136,12 @@ impl CmdArgs {
     }
 }
 
+impl Default for CmdArgs {
```

Comment:
> Do we wind up calling this anywhere, or is this just defined because we can?

No, it's only defined here and never called so far.

> I'm not sure how I feel about having both new and default doing the same thing; should we remove new and just call default maybe?

This is arguably one of the more controversial clippy lints. It mirrors the conventions in `std`, where usually both `T::new` and `Default for T` are implemented. Having `Default` is mostly useful for generic APIs, such as `std::mem::take` (encouraged by another lint), so it is "future proofing" in a way.

Do you have a preference? We could easily remove the `Default` implementations until they are needed and disable the lint.

Reply:
Thanks for the explanation. I think I'd prefer reverting these. When we get around to releasing some of the crates (like `widgetry` especially) on crates.io eventually, then it makes more sense to future-proof and standardize the APIs more. But in the meantime, unused methods seem odd. There are maybe cases when calling `Foo::default()` "feels" more normal than `Foo::new()`, but definitely more where `new` reads better. For instance, `GeomBatch::new()` vs `GeomBatch::default()`.

Reply:
Done.